### DISCUSSION TITLE
Interface 'ErrnoException' incorrectly extends interface 'Error' with exactOptionalPropertyTypes enabled

### DISCUSSION THREAD

**Discussion by techmunk** [DefinitelyTyped/DefinitelyTyped#54673] [Issues with a @types package]
Ts 4.4@next, building with `exactOptionalPropertyTypes` causes this error:

```
node_modules/@types/node/globals.d.ts:172:15 - error TS2430: Interface 'ErrnoException' incorrectly extends interface 'Error'.
  Types of property 'stack' are incompatible.
    Type 'string | undefined' is not assignable to type 'string'.
      Type 'undefined' is not assignable to type 'string'.

172     interface ErrnoException extends Error {
                  ~~~~~~~~~~~~~~


Found 1 error.
```

Changing stack property to match declaration in typescripts `lib.es5.d.ts` resolves this issue.

I believe this was added in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54342 in preparation for the next typescript release.

This is the only issue I've noticed building with exactOptionalPropertyTypes enabled so far. Though my current test case is very limited.

**Comment by elektronik2k5** [comment]
I'm facing the same issue. Managed to work around it using [patch-package](https://github.com/ds300/patch-package) with versions `16.7.3`, `16.7.4`, `16.7.6`, `16.7.7` and `16.7.10` of `@types/node` using this patch:

```diff
diff --git a/node_modules/@types/node/globals.d.ts b/node_modules/@types/node/globals.d.ts
index 5e359db..526273f 100755
--- a/node_modules/@types/node/globals.d.ts
+++ b/node_modules/@types/node/globals.d.ts
@@ -185,7 +185,8 @@ declare namespace NodeJS {
         code?: string | undefined;
         path?: string | undefined;
         syscall?: string | undefined;
-        stack?: string | undefined;
+        // Workaround till this issue is fixed: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/54673
+        stack?: string;
     }
 
     interface ReadableStream extends EventEmitter {
```

I haven't really considered any implications of this change but now TS compiles just fine.

**Comment by fgblomqvist** [comment]
Updating`@types/node` to latest solved this for me.
